webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -761,6 +762,11 @@ func (v validator) validateReadinessProbe(ctx *context.WebhookRequestContext, vm
 
 	readinessProbePath := field.NewPath("spec", "readinessProbe")
 
+	if networkType := pkgconfig.FromContext(ctx).NetworkProviderType; networkType == pkgconfig.NetworkProviderTypeVPC {
```

Comment:
Only the `TCPProbe` needs to be disabled in VPC because of the network restrictions. The others are fine